src/core/classes/language.ts

```diff
@@ -11,7 +11,7 @@ export default class Language extends EventTarget {
 	registry: LanguageRegistry;
 	resolvedGrammar?: Grammar;
 	require: List<LanguageLike> = new List();
-	optional: List<string> = new List();
+	optional: List<string | LanguageLike> = new List();
```

Comment:
We add languages from `extends` to this list, and `extends` has a broader type we should match.